homeassistant/components/ssdp/__init__.py

```diff
@@ -663,11 +663,12 @@ async def _async_find_next_available_port(source: AddressTupleVXType) -> int:
     test_socket.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
 
     for port in range(UPNP_SERVER_MIN_PORT, UPNP_SERVER_MAX_PORT):
+        addr = (source[0],) + (port,) + source[2:]
         try:
-            test_socket.bind(source)
+            test_socket.bind(addr)
             return port
         except OSError:
-            if port == UPNP_SERVER_MAX_PORT:
+            if port == UPNP_SERVER_MAX_PORT - 1:
```

Comment:
Is this correct, or is the range wrong? Maybe it doesn't matter?

Reply:
`range()` is exclusive regarding the `stop` value. I.e., `range(UPNP_SERVER_MIN_PORT, UPNP_SERVER_MAX_PORT)` results in `UPNP_SERVER_MIN_PORT..UPNP_SERVER_MAX_PORT - 1`.

When the last port is tested, the `port` value is `UPNP_SERVER_MAX_PORT - 1`. If the port is not free, we want to re-raise the `OSError`.